Release 9 release notes draft.

garloff opened the pull request:
This should be mostly complete w.r.t. Cluster-Stacks, OSISM-10 details still missing.

garloff committed:
Release 9 release notes draft.

This should be mostly complete w.r.t. Cluster-Stacks, OSISM-10 details
still missing.

Signed-off-by: Kurt Garloff <kurt@garloff.de>

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+stack series ["scs2"](https://github.com/SovereignCloudStack/cluster-stacks/tree/main/providers/openstack/scs2)
+starts with [Kubernetes 1.33.x](https://kubernetes.io/releases/)
+and leverages cluster-api v1.10 and cluster-api-provider-openstack v0.12
+as well as the matching cilium, cinder-csi and OCCM versions.
```
```suggestion
The node images are also built with Ubuntu 24.04 LTS now. The new cluster stack ["scs2"](https://github.com/SovereignCloudStack/cluster-stacks/tree/main/providers/openstack/scs2)
starts with [Kubernetes 1.33.x](https://kubernetes.io/releases/)
and leverages Cluster API v1.10 and Cluster API Provider OpenStack v0.12
as well as the matching Cilium, Cinder CSI and OCCM versions.
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+
+~...TODO... OSISM testbed, CiaB~
+
+The new `openstack-scs2` cluster class series which succeeds `openstack-scs`
```
```suggestion
The new `scs2` cluster stack which succeeds the `scs` cluster stack
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+Rather than generating the secrets in two different formats (which was done
+using a [helper helm chart](https://github.com/SovereignCloudStack/openstack-csp-helper)
+previously), we now use only one secret, simplifying
+the handling by only requiring one clusterResourceSet to be managed.
```
```suggestion
the handling by only requiring one ClusterResourceSet to be managed.
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+This also allows to support self-signed certificates (a custom CA for
+the OpenStack API) without trouble.
+
+The [clusterClass variables](https://github.com/SovereignCloudStack/cluster-stacks/blob/f5f9a4260d32bac33ff87146cb1f88ac55288bc9/providers/openstack/scs2/cluster-class/templates/cluster-class.yaml#L35)
```
```suggestion
The [ClusterClass variables](https://github.com/SovereignCloudStack/cluster-stacks/blob/f5f9a4260d32bac33ff87146cb1f88ac55288bc9/providers/openstack/scs2/cluster-class/templates/cluster-class.yaml#L35)
```

garloff replied:
I was wondering: Do we have a doc somewhere where the variables are documented outside of the clusterClass yaml? That would be the best place to link then ...

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+consistent camelCase naming scheme. The defaults now use diskless flavors,
+providing a better preconfiguration on SCS virtualization infrastructure.
+Loadbalancers, flavors, disks etc. can be configured using clusterClass
+variables. The clusterClass releases no longer bundle the helm charts
```
```suggestion
Load balancers, flavors, disks etc. can be configured using ClusterClass
variables. The ClusterClass releases no longer bundle the Helm charts
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+providing a better preconfiguration on SCS virtualization infrastructure.
+Loadbalancers, flavors, disks etc. can be configured using clusterClass
+variables. The clusterClass releases no longer bundle the helm charts
+for the clusterAddons (such as CNI, CSI, CCM, metrics). Instead, they
```
```suggestion
for the cluster addons (such as CNI, CSI, CCM, metrics). Instead, they
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+by the Cluster Stack Operator (CSO). A new version v0.2.0-alpha.1 of
+CSO has been released to support this -- it also still supports the old
+way with bundled helm charts. In either case, the version is pinned by
+the specific clusterClass, so users remain on validated paths.
```
```suggestion
the specific ClusterClass, so users remain on validated paths.
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+that the upstream OpenStack community does to support the so-called
+Skip-Level-Upgrade-Release-Process ([SLURP](https://docs.openstack.org/project-team-guide/release-cadence-adjustment.html)).
+
+The new cluster stack series "scs2" can live next to existing "scs"
```
```suggestion
The new cluster stack "scs2" can live next to existing "scs"
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+possible. The code to do this conversion automatically is still
+in development and will be carefully tested before being released.
+
+The scs2 clusters support Kubernetes 1.33.x and future K8s versions.
```
```suggestion
The scs2 clusters support Kubernetes 1.33.x and future Kubernetes versions.
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+The scs2 clusters support Kubernetes 1.33.x and future K8s versions.
+At release time, the cluster stack `openstack-scs2-1-33-v1` containing
+k8s-v1.33.4 was current.
+The old scs series will end with the final 1.31.x and 1.32.x patch
```
```suggestion
The old `scs` cluster stack will end with the final 1.31.x and 1.32.x patch
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+no longer maintained. The community is working on documentation
+and/or code to provide guidance to operators how to perform the tests.
+
+The k8s clusters created with the scs and scs2 cluster stacks
```
```suggestion
The Kubernetes clusters created with the scs and scs2 cluster stacks
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+pass the [CNCF e2e conformance tests](https://github.com/cncf/k8s-conformance),
+with the exception of the `HostPort validates that there is no conflict
+between pods with same hostPort but different hostIP and protocol`
+test, which is not supported with the our default cilium CNI.
```
```suggestion
test, which is not supported with our default cilium CNI.
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+As mentioned above, the migration script and docs are still in
+development for conversion of `scs` to `scs2` cluster stacks.
+
+Due to the way that the helm chart of the OCCM currently creates
```
```suggestion
Due to the way that the Helm chart of the OCCM currently creates
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+
+Due to the way that the helm chart of the OCCM currently creates
+the `cloud.conf` configuration file, it is not straightforward
+to control the type of Loadbalancer that OCCM creates for the
```
```suggestion
to control the type of load balancer that OCCM creates for the
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+the `cloud.conf` configuration file, it is not straightforward
+to control the type of Loadbalancer that OCCM creates for the
+cluster upon demand. So, currently, the cloud's default provider
+will be used. We wanted to add a clusterClass variable that
```
```suggestion
will be used. We wanted to add a ClusterClass variable that
```

jschoone commented on Release9.md:
```diff
@@ -0,0 +1,182 @@
+under discussion. This will be delivered with a future update
+and will not be a breaking change.
+ 
+Currently, when creating clusters, the successful start of coredns
```
```suggestion
Currently, when creating clusters, the successful start of CoreDNS
```

garloff committed:
Apply suggestions from code review

Removed series designation from the new `scs2` ClusterClass series. Hmmm, should we discuss and clarify this once more?

Co-authored-by: Jan Schoone <6106846+jschoone@users.noreply.github.com>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Apply suggestions from code review

Capitalization and separate words.

Co-authored-by: Jan Schoone <6106846+jschoone@users.noreply.github.com>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Apply suggestions from code review

k8s -> Kubernetes

Co-authored-by: Jan Schoone <6106846+jschoone@users.noreply.github.com>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff committed:
Typo (extra "the")

Co-authored-by: Jan Schoone <6106846+jschoone@users.noreply.github.com>
Signed-off-by: Kurt Garloff <kurt@garloff.de>

garloff commented:
Thanks a lot, @jschoone.

I think I'd still like to discuss how we differentiate a Cluster Stack `openstack-scs2-v1-33-v1` from version series `openstack-scs2-v1-33-*` from a series `openstack-scs2-*`. Naming them all the same will confuse ourselves.
See https://github.com/SovereignCloudStack/sovereigncloudstack.org/pull/65#issuecomment-3340538941.

Despite this, I have merged all your improvements, including the ones that [removed the "series"](https://github.com/SovereignCloudStack/sovereigncloudstack.org/pull/65#issuecomment-3340538941). I split the commits into groups though, so we can easily revert depending on the outcome of the discussion.

jschoone commented:
> Thanks a lot, @jschoone.
> 
> I think I'd still like to discuss how we differentiate a Cluster Stack `openstack-scs2-v1-33-v1` from version series `openstack-scs2-v1-33-*` from a series `openstack-scs2-*`. Naming them all the same will confuse ourselves. See [SovereignCloudStack/sovereigncloudstack.org#65 (comment)](https://github.com/SovereignCloudStack/sovereigncloudstack.org/pull/65#issuecomment-3340538941).
> 
> Despite this, I have merged all your improvements, including the ones that [removed the "series"](https://github.com/SovereignCloudStack/sovereigncloudstack.org/pull/65#issuecomment-3340538941). I split the commits into groups though, so we can easily revert depending on the outcome of the discussion.

Sorry somehow I completely missed the discussion on that. If calling it "series" makes somehow sense to you and to others you can keep it. 
For me personally it's more confusing because this introduces new wording which creates the impression of new concepts. So this probably needs to end up in the glossary as well.

A cluster stack (e.g. `scs2`) should have a fixed set of addons and features which you can expecte to work on the providers for what the cluster stack is implement. Maybe it can be compared to another software, like Chromium. You can run Chromium on different operating systems, where probably no one says linux-chromium or windows-chromium, but everywhere you expect the same features and behavior. Also it has an engine with an own version (= kubernetes version) and a version for itself (= cluster stack version). Then there are other browsers with the same engine but called Brave or Vivaldi because of some other features. People will say they use the browser Brave, Vivaldi or Chromium and not the "browser series". With those information you can read up which features the browser will get you and you can use them of the operating system (= provider) of your choice.
Not sure if that comparison works, to me it sounds logical :D

garloff committed:
Merge branch 'main' into feat/release-notes-9

garloff commented:
If we call `scs2` a Cluster Stack, then we should should have different names for `scs2-1-33` (maybe "Cluster Stack minor releases"?) and `scs2-1-33-v1` (maybe "Cluster Stack release"?).

No strong opinion here ... I intuitively had called `scs2-1-33-v1` a Cluster Stack and then we'd need qualifiers for the others. From looking at the way our syself friends use it, is there a clear preference?

fkr commented:
@berendt I was approched today again regarding the OSISM-10 that is supposedly part of R9 (and also about the R9 not being visible as well).

garloff commented:
Any updates? @berendt ?

fkr commented:
What is going to happen with the release notes for R9? I noticed this pending PR today while updating things elsewhere. @SovereignCloudStack/project-board @berendt
